exploration/train_il_agent.py

```diff
@@ -0,0 +1,101 @@
+from typing import List, Tuple
+
+import gymnasium as gym
+import imitation.algorithms.bc
+import numpy as np
+from clearml import Task
+from imitation.data import rollout
+from imitation.data.wrappers import RolloutInfoWrapper
+from imitation.policies.serialize import load_policy
+from stable_baselines3.common.env_util import DummyVecEnv
+from stable_baselines3.common.monitor import Monitor
+
+from rl_framework.agent.imitation import ImitationAgent
+from rl_framework.util import (
+    ClearMLConnector,
+    ClearMLDownloadConfig,
+    ClearMLUploadConfig,
+)
+
+
+def sample_expert_sequences(
+    environment, timesteps
+) -> List[List[Tuple[object, object, object, float, bool, bool, dict]]]:
+    def download_expert_policy(venv):
+        policy = load_policy(
+            "ppo-huggingface",
+            organization="HumanCompatibleAI",
+            env_name="seals-CartPole-v0",
+            venv=venv,
+        )
+        return policy
+
+    environment_return_functions = [lambda: RolloutInfoWrapper(Monitor(environment))]
+    vectorized_environment = DummyVecEnv(env_fns=environment_return_functions)
+    expert_policy = download_expert_policy(vectorized_environment)
+
+    rollouts = rollout.rollout(
+        expert_policy,
+        vectorized_environment,
+        rollout.make_sample_until(min_timesteps=timesteps),
+        rng=np.random.default_rng(0),
+    )
+
+    episode_sequences = []
+    for trajectory in rollouts:
```

Comment:
This conversion could be the job of the discussed dataset wrapper (e.g., from_buffer(rollouts).iterate_tupple_trajectories())

Reply:
Yes, this will be implemented as a functionality in the dataset class.